pkg/controller/networkpolicy/validate.go

```diff
@@ -213,6 +214,33 @@ func (v *NetworkPolicyValidator) validateAntreaPolicy(curObj, oldObj interface{}
 	return reason, allowed
 }
 
+// validatePort validates if ports is valid
+func (a *antreaPolicyValidator) validatePort(ingress, egress []secv1alpha1.Rule) error {
+	isValid := func(rules []secv1alpha1.Rule) error {
+		for _, rule := range rules {
+			for _, port := range rule.Ports {
```

Comment:
maybe it will be helpful to do a quick check on 
```
if port.Endport == nil {
    continue
}
```
and then perform the validations without having to check for endport everytime

Reply:
Good point.